js/simpletextbox/Index.svelte

```diff
@@ -0,0 +1,167 @@
+<svelte:options accessors={true} />
+
+<script lang="ts">
+	import type { Gradio, SelectData } from "@gradio/utils";
+	import { BlockTitle } from "@gradio/atoms";
+	import { Block } from "@gradio/atoms";
+	import { StatusTracker } from "@gradio/statustracker";
+	import type { LoadingStatus } from "@gradio/statustracker";
+	import { tick } from "svelte";
+
+	export let gradio: Gradio<{
+		change: string;
+		submit: never;
+		blur: never;
+		select: SelectData;
+		input: never;
+		focus: never;
+	}>;
+	export let label = "Textbox";
+	export let elem_id = "";
+	export let elem_classes: string[] = [];
+	export let visible = true;
+	export let value = "";
+	export let placeholder = "";
+	export let show_label: boolean;
+	export let scale: number | null = null;
+	export let min_width: number | undefined = undefined;
+	export let loading_status: LoadingStatus | undefined = undefined;
+	export let value_is_output = false;
+	export let mode: "static" | "interactive";
+	export let rtl = false;
+
+
+	let el: HTMLTextAreaElement | HTMLInputElement;
+	const container = true;
+
+	function handle_change(): void {
+		gradio.dispatch("change", value);
+		if (!value_is_output) {
+			gradio.dispatch("input");
+		}
+	}
+
+	function handle_select(event: Event): void {
+		const target: HTMLTextAreaElement | HTMLInputElement = event.target as
+			| HTMLTextAreaElement
+			| HTMLInputElement;
+		const text = target.value;
+		const index: [number, number] = [
+			target.selectionStart as number,
+			target.selectionEnd as number,
+		];
+		gradio.dispatch("select", { value: text.substring(...index), index: index });
+	}
+
+	async function handle_keypress(e: KeyboardEvent): Promise<void> {
+		await tick();
+		if (e.key === "Enter") {
+			e.preventDefault();
+			gradio.dispatch("submit");
+		}
+	}
+
+	$: if (value === null) value = "";
+
+	$: value, handle_change();
+
+
+</script>
+
+<Block
+	{visible}
+	{elem_id}
+	{elem_classes}
+	{scale}
+	{min_width}
+	allow_overflow={false}
+	padding={true}
+>
+	{#if loading_status}
+		<StatusTracker
+			autoscroll={gradio.autoscroll}
+			i18n={gradio.i18n}
+			{...loading_status}
+		/>
+	{/if}
+
+	<label class:container>
+		<BlockTitle {show_label} info={undefined}>{label}</BlockTitle>
+
+		<input
+			data-testid="textbox"
+			type="text"
+			class="scroll-hide"
+			bind:value
+			bind:this={el}
+			{placeholder}
+			disabled={mode === "static"}
+			dir={rtl ? "rtl" : "ltr"}
+			on:keypress={handle_keypress}
+			on:blur={() => gradio.dispatch("blur")}
+			on:select={handle_select}
+			on:focus={() => gradio.dispatch("focus")}
```

Comment:
As above i think we could get rid of some of these and it would help keep things streamlined.